staging/src/k8s.io/kubectl/pkg/cmd/run/run.go

```diff
@@ -166,6 +166,20 @@ func NewCmdRun(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Co
 	addRunFlags(cmd, o)
 	cmdutil.AddApplyAnnotationFlags(cmd)
 	cmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodAttachTimeout)
+
+	// Deprecate and hide unused flags.
+	// These flags are being added to the run command by DeleteFlags to support pod deletion after attach,
+	// but they are not used if set, so they effectively do nothing.
```

Comment:
It looks like the values are passed through during `Complete`, so I was wondering how it is that they don't actually work. Does the following match your understanding?
- `wait` and `grace-period` are overridden at L235-235 to `false` and `-1` respectively
- `timeout` is always ignored when `wait=false`
- `filenames`, `kustomize` and `recursive` would have to be passed to the Builder created at L425 via `FilenameParam`, but they aren't. So when `DeleteOptions#DeleteResult` runs, the resources iterated over are only the ones given by `Builder#ResourceNames`, i.e. the created objects (good!).
- `force` under the hood actually turns into `grace-period=0`, so our overriding it to `-1` as stated above nullfies this one too. The grace period would show up in the delete request body with -v=8 if this weren't true (it doesn't).

⚠️ I think `cascade` technically does work. If I do `kubectl run -i -t mypod --image=busybox:latest -f base/resources.yaml --rm --pod-running-timeout=1s --cascade=orphan -v=8`, then the object I get back includes `"finalizers":["orphan"]`. That said, this doesn't strike me as useful, since pods don't have dependents by default and this command in any case doesn't create any. But if we do remove it as well, we should use different reasoning. Wdyt?

Reply:
@KnVerey thanks for tracing through all that.  I should have done that with this PR description!

I agree on all points.  

Also, I'm not sure why I thought cascade was a no-op, but you're right, technically it is used.  I'm not sure what effect it would have in the case of `kubectl run` (there would be no dependent resources?) but the reason for the deprecation is not the same as the other flags.

I'll break it out from the others, with a different comment, and different deprecation message... maybe just `"This flag will be removed in version 1.29"`. 

I'm also not opposed to leaving it, if there is a valid use case. I just can't think of one offhand or how we would explain to people when they should use it.

Reply:
Yeah I can't really think of one either. Technically you could manually add ownerReferences to objects to cause it to have dependents, but I have no idea why you'd do that for an object created with `run`. Of course, removing the flag doesn't prevent them from doing that, only from changing the deletion policy used with the `rm` option--and `rm` can only be specified when `run` is used interactively (making it extra unlikely this is useful). If we want a reason, maybe "is not relevant for kubectl run"?